### PR TITLE
Modify wps_scanner to first connect to a known network

### PR DESCRIPTION
**Pull request by stefanos1316**
Issue [#67](https://github.com/eellak/epidose/issues/67)

### CODE REVIEW

File: epidose/device/wps_scanner_d.sh [hunk 1]
```diff
@@ -112,17 +110,110 @@ wps_connect()
...
+{
+  # Check if network $1 is in range
+  get_network_if_in_range=$(wpa_cli -i wlan0 scan_results | grep "$1")
+  if [ -z "$get_network_if_in_range"  ]; then
+    log "Network $1 not in range"
+    return 1
+  fi
+
+  # Get $1 network's id, list_networks checks the /etc/wpa_supplicant/wpa_supplicant.conf file
+  get_network_id=$(wpa_cli list_networks | grep "$1" | cut -f 1)
```
Comment (dspinellis) [inline epidose/device/wps_scanner_d.sh]: * Replace grep/cut with `awk '$2 ~ /^'$1'$/ {print $1}'` to ensure better matching.
* Remove `get_` from variable names: variables contain values they don't perform actions

File: epidose/device/wps_scanner_d.sh [hunk 2]
```diff
@@ -112,17 +110,110 @@ wps_connect()
...
+  if [ -z "$get_network_id" ]; then
+    log "Network $1 configurations may be missing from wpa_supplicant.conf"
+    return 1
```
Comment (dspinellis) [inline epidose/device/wps_scanner_d.sh:153]: … not found … is more clear/.

File: epidose/device/wps_scanner_d.sh [hunk 3]
```diff
@@ -112,17 +110,110 @@ wps_connect()
...
+  return 0
+}
+
+# Checks if the device can connect
+# first to Eduroam, then to Epidose backup,
+# and then to an WPS-supported network
+# precondition: none
+# Returns 1 if the device failed
+# to connect to any of the networks
+select_network()
```
Comment (dspinellis) [inline epidose/device/wps_scanner_d.sh]: Better name `try_alternative_wifi_connections`.

File: epidose/device/wps_scanner_d.sh [hunk 4]
```diff
@@ -112,17 +110,110 @@ wps_connect()
...
+{
+  log "Trying to connect to eduroam"
```
Comment (dspinellis) [inline epidose/device/wps_scanner_d.sh]: `for i in eduroam epidose ; do` …

**Review by dspinellis — CHANGES_REQUESTED** [review]
Looking good overall, but some changes are needed.

**Comment by stefanos1316** [comment]
Thank you for your comments.
I have submitted a commit with the requested changes.

**Pull request merged**